From 30031a79ed93a855df797467afed4d0c1ed03de7 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Tue, 19 May 2026 08:12:03 +0000
Subject: [PATCH 050/210] fixup! drm/meson: encoder_hdmi: fix undefined
 err_put_node label

Commit df4fc5788130 ("FROMLIST(v1): drm/meson: encoder_hdmi: Use bridge
connector CEC notifier") removed the err_put_node label and moved
of_node_put(remote) earlier in meson_encoder_hdmi_probe(), but left
several goto err_put_node statements behind. The kernel fails to build
with:

  drivers/gpu/drm/meson/meson_encoder_hdmi.c: In function 'meson_encoder_hdmi_probe':
  drivers/gpu/drm/meson/meson_encoder_hdmi.c:395:17: error: label 'err_put_node' used but not defined

At each remaining call site the remote device_node has already been
released, so convert the error paths to return dev_err_probe(...)
directly, matching the pattern used for the next_bridge failure.

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 drivers/gpu/drm/meson/meson_encoder_hdmi.c | 16 +++++-----------
 1 file changed, 5 insertions(+), 11 deletions(-)

diff --git a/drivers/gpu/drm/meson/meson_encoder_hdmi.c b/drivers/gpu/drm/meson/meson_encoder_hdmi.c
index 39b35334984e..b233e39a33b6 100644
--- a/drivers/gpu/drm/meson/meson_encoder_hdmi.c
+++ b/drivers/gpu/drm/meson/meson_encoder_hdmi.c
@@ -370,30 +370,24 @@ int meson_encoder_hdmi_probe(struct meson_drm *priv)
 	/* Encoder */
 	ret = drm_simple_encoder_init(priv->drm, &meson_encoder_hdmi->encoder,
 				      DRM_MODE_ENCODER_TMDS);
-	if (ret) {
-		dev_err_probe(priv->dev, ret, "Failed to init HDMI encoder\n");
-		goto err_put_node;
-	}
+	if (ret)
+		return dev_err_probe(priv->dev, ret, "Failed to init HDMI encoder\n");
 
 	meson_encoder_hdmi->encoder.possible_crtcs = BIT(0);
 
 	/* Attach HDMI Encoder Bridge to Encoder */
 	ret = drm_bridge_attach(&meson_encoder_hdmi->encoder, &meson_encoder_hdmi->bridge, NULL,
 				DRM_BRIDGE_ATTACH_NO_CONNECTOR);
-	if (ret) {
-		dev_err_probe(priv->dev, ret, "Failed to attach bridge\n");
-		goto err_put_node;
-	}
+	if (ret)
+		return dev_err_probe(priv->dev, ret, "Failed to attach bridge\n");
 
 	/* Initialize & attach Bridge Connector */
 	meson_encoder_hdmi->connector = drm_bridge_connector_init(priv->drm,
 							&meson_encoder_hdmi->encoder);
-	if (IS_ERR(meson_encoder_hdmi->connector)) {
+	if (IS_ERR(meson_encoder_hdmi->connector))
 		ret = dev_err_probe(priv->dev,
 				    PTR_ERR(meson_encoder_hdmi->connector),
 				    "Unable to create HDMI bridge connector\n");
-		goto err_put_node;
-	}
 
 	/*
 	 * We should have now in place:
-- 
2.43.0

